### PR TITLE
Remove normed keyword in histogram helper functions

### LINKED ISSUE #8963: Deprecate "normed" kwarg from histogram function helpers
The `normed` kwarg has been deprecated both in mpl and numpy for a while, thus I suggest we also deprecate it from astropy/units/quantity_helper/function_helpers.py, from function helpers `histogram`, `histogram2d`, and `histogramdd`

### PR DESCRIPTION
**Pull request by pllim**
Fix #8963

### CODE REVIEW

File: astropy/units/quantity_helper/function_helpers.py [hunk 1]
```diff
@@ -584,9 +585,9 @@ def _check_bins(bins, unit):
         return bins
 
 
+@deprecated_renamed_argument('normed', 'density', '4.0')
```
Comment (pllim) [inline astropy/units/quantity_helper/function_helpers.py]: Is this a proper way to deprecate `normed`? Seems like this is some `ufunc` magic, so I am not sure.

Comment (bsipocz) [reply astropy/units/quantity_helper/function_helpers.py]: yes, this should work, but let's see the tests follow the theory, too.

Comment (bsipocz) [reply astropy/units/quantity_helper/function_helpers.py]: however, given that density was already a kwarg, you could simply use it without the alternative part to just remove `normed`.

Comment (pllim) [reply astropy/units/quantity_helper/function_helpers.py]: I was trying to find example of something like `@deprecated_renamed_argument('normed', None, '4.0')` but I can't find anything; I don't think that use case is even tested. Also, setting second arg to `None` also means `normed` check is still needed in the function, so more work down the line when it is time to completely remove it. 🤷‍♀

Comment (pllim) [reply astropy/units/quantity_helper/function_helpers.py]: Well, well, this broke numpy-dev anyway. So, I'll try your suggestion and see. https://travis-ci.org/astropy/astropy/jobs/561161578

Also, the codecov failure makes no sense to me.

**Comment by mhvk** [comment]
Since this didn't previously work at all, I think we could in principle outright remove it. 

Though since we are just overriding numpy, maybe easiest is just be consistent with them? Indeed, doesn't one get a deprecation warning from numpy anyway?

**Comment by pllim** [comment]
@mhvk , consistent in what way? Can you please clarify? Do I have to code dive in Numpy-land and copy-paste something over?

File: astropy/units/quantity_helper/function_helpers.py [hunk 2]
```diff
@@ -584,9 +585,9 @@ def _check_bins(bins, unit):
         return bins
 
 
+@deprecated_renamed_argument('normed', None, '4.0')
```
Comment (pllim) [inline astropy/units/quantity_helper/function_helpers.py]: Is this better?

**Comment by mhvk** [comment]
No need to dive into numpy :raised_hand_with_fingers_splayed: 

I just wondered to what extent we need this at all, since one presumably gets a deprecation warning from numpy anyway. So, can we not just stick with the numpy signature (as is) and remove `normed` without any deprecation when they remove it? 

Alternatively, we could simply not have it at all already (which perhaps is what @bsipocz was thinking in #8963).

**Comment by pllim** [comment]
Okay, so I am hearing two different approaches here:

1. Leave it as is in `master` and let Numpy emit the warning. Delete it when Numpy does (which means we cannot fully delete it until our minversion of Numpy is the Numpy that removed it).
2. Straight out delete `normed` in helper functions without any deprecation on our side.

How should I proceed? Please advise.

**Comment by bsipocz** [comment]
> I just wondered to what extent we need this at all, since one presumably gets a deprecation warning from numpy anyway. So, can we not just stick with the numpy signature (as is) and remove normed without any deprecation when they remove it?

I  suppose you're right. From this viewpoint, this as a case when we are the users of a deprecated functionality, so removing the kwarg straight away would be a good solution.

File: astropy/units/quantity_helper/function_helpers.py [hunk 3]
```diff
@@ -598,10 +597,10 @@ def histogram(a, bins=10, range=None, normed=None, weights=None,
     if not isinstance(bins, str):
         bins = _check_bins(bins, a.unit)
 
-    if density or normed:
+    if density:
         unit = (unit or 1) / a.unit
 
-    return ((a.value, bins, range, normed, weights, density), {},
+    return ((a.value, bins, range, None, weights, density), {},
```
Comment (mhvk) [inline astropy/units/quantity_helper/function_helpers.py]: To future-proof this for the eventual removal of `normed`, let's do `((a.value, bins, range), {'weights': weights, 'density': density}, ...` (here and below)

File: CHANGES.rst [hunk 4]
```diff
@@ -244,6 +244,9 @@ astropy.units
   where numpy scalars behave differently than python ones (e.g., taking the
   square root of a negative number). [#8876]
 
+- Removed ``normed`` keyword for histogram functions in ``function_helpers``.
```
Comment (mhvk) [inline CHANGES.rst]: Don't need a change-log entry - this is `affects-dev`...

File: astropy/units/quantity_helper/function_helpers.py (hunk 3 above)
Comment (pllim) [reply astropy/units/quantity_helper/function_helpers.py]: Ahhh, so that's how it hooks up to Numpy API. Gotcha. Done!

File: CHANGES.rst (hunk 4 above)
Comment (pllim) [reply CHANGES.rst]: Okay. Removed.

Comment (bsipocz) [reply CHANGES.rst]: Oh, I missed that this is affects-dev, then the changes look even better.

**Comment by mhvk** [comment]
Of course "when ci passes" includes numpy-dev...

**Comment by bsipocz** [comment]
And it doesn't, this one looks relevant:
```
=================================== FAILURES ===================================
___________________ TestHistogramFunctions.test_histogram2d ____________________
self = <astropy.units.tests.test_quantity_non_ufuncs.TestHistogramFunctions object at 0x7f2a2656d5c0>
    @pytest.mark.xfail(NO_ARRAY_FUNCTION,
                       reason="Needs __array_function__ support")
    def test_histogram2d(self):
        x, y = self.x, self.y
        weights = self.weights
        # Basic tests with X, Y.
        self.check(np.histogram2d, x, y,
                   value_args=(x.value, y.value),
                   expected_units=(None, x.unit, y.unit))
        # Check units with density.
        self.check(np.histogram2d, x, y, density=True,
                   value_args=(x.value, y.value),
                   expected_units=(1/(x.unit*y.unit), x.unit, y.unit))
        # Check units with weights.
        self.check(np.histogram2d, x, y, weights=weights,
                   value_args=(x.value, y.value),
                   value_kwargs=dict(weights=weights.value),
                   expected_units=(weights.unit, x.unit, y.unit))
        # Check quantity bin sizes.
        inb_y = [0, 0.025, 1.] * u.m
        self.check(np.histogram2d, x, y, [5, inb_y],
                   value_args=(x.value, y.value,
                               [5, np.array([0, 2.5, 100.])]),
                   expected_units=(None, x.unit, y.unit))
        # Check we dispatch on bin sizes (and check kwarg as well).
        inb2_y = [0, 250, 10000.] * u.percent
        self.check(np.histogram2d, x.value, y.value, bins=[5, inb2_y],
                   value_args=(x.value, y.value),
                   value_kwargs=dict(bins=[5, np.array([0, 2.5, 100.])]),
                   expected_units=(None, u.one, u.one))
    
        with pytest.raises(u.UnitsError):
>           np.histogram2d(x, y, [125, 200] * u.s)
astropy/units/tests/test_quantity_non_ufuncs.py:1395: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
<__array_function__ internals>:6: in histogram2d
    ???
astropy/units/quantity.py:1511: in __array_function__
    result = super().__array_function__(function, types, args, kwargs)
/home/travis/miniconda/envs/test/lib/python3.7/site-packages/numpy/lib/twodim_base.py:715: in histogram2d
    hist, edges = histogramdd([x, y], bins, range, normed, weights, density)
<__array_function__ internals>:6: in histogramdd
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
self = <Quantity 125. s>, function = <function histogramdd at 0x7f2a59f8b2f0>
types = (<class 'astropy.units.quantity.Quantity'>, <class 'numpy.ndarray'>)
args = ([array([1.1, 1.2, 1.3, 2.1, 5.1]), array([1.2, 2.2, 2.4, 3. , 4. ])], [<Quantity 125. s>, <Quantity 200. s>], None, None, None, None)
kwargs = {}
    def __array_function__(self, function, types, args, kwargs):
        """Wrap numpy functions, taking care of units.
    
        Parameters
        ----------
        function : callable
            Numpy function to wrap
        types : iterable of classes
            Classes that provide an ``__array_function__`` override. Can
            in principle be used to interact with other classes. Below,
            mostly passed on to `~numpy.ndarray`, which can only interact
            with subclasses.
        args : tuple
            Positional arguments provided in the function call.
        kwargs : dict
            Keyword arguments provided in the function call.
    
        Returns
        -------
        result: `~astropy.units.Quantity`, `~numpy.ndarray`
            As appropriate for the function.  If the function is not
            supported, `NotImplemented` is returned, which will lead to
            a `TypeError` unless another argument overrode the function.
    
        Raises
        ------
        ~astropy.units.UnitsError
            If operands have incompatible units.
        """
        # A function should be in one of the following sets of dicts:
        # 1. SUBCLASS_SAFE_FUNCTIONS (set), if the numpy implementation
        #    supports Quantity; we pass on to ndarray.__array_function__.
        # 2. FUNCTION_HELPERS (dict), if the numpy implementation is usable
        #    after converting quantities to arrays with suitable units,
        #    and possibly setting units on the result.
        # 3. DISPATCHED_FUNCTIONS (dict), if the function makes sense but
        #    requires a Quantity-specific implementation.
        # 4. UNSUPPORTED_FUNCTIONS (set), if the function does not make sense.
        # For now, since we may not yet have complete coverage, if a
        # function is in none of the above, we simply call the numpy
        # implementation.
        if function in SUBCLASS_SAFE_FUNCTIONS:
            return super().__array_function__(function, types, args, kwargs)
    
        elif function in FUNCTION_HELPERS:
            function_helper = FUNCTION_HELPERS[function]
            try:
>               args, kwargs, unit, out = function_helper(*args, **kwargs)
E               TypeError: histogramdd() takes from 1 to 5 positional arguments but 6 were given
astropy/units/quantity.py:1507: TypeError
====== 1 failed, 13146 passed, 113 skipped, 105 xfailed in 356.49 seconds ======
The command "$MAIN_CMD $SETUP_CMD" exited with 1.
```

**Comment by mhvk** [comment]
Darn, that looks like numpy itself calls it with a `normed` argument. Though it is in its `histogram2d` implementation and we should never get there - as nothing should be a `Quantity` any more... So, a bug in my override...

**Comment by pllim** [comment]
@mhvk , what is the proposed fix for the numpy dev failure? Please advise.

**Comment by mhvk** [comment]
I think fixing the small bug in my implementation will make things pass. Hope to push something to your PR later today.

**Comment by mhvk** [comment]
Was a bit trickier than I thought, but hopefully it will now be OK (well, it should - tests pass locally). Good to see that this smoked out bugs! (I fear there will be more...)

**Comment by bsipocz** [comment]
> Good to see that this smoked out bugs! (I fear there will be more...)

The advantage to get new features merged in early in the release cycle, there are still plenty opportunity to fix things before it lands with actual users.

**Comment by bsipocz** [comment]
This is all green now (the numpy dev job is failing due to a fluke remote data failure, totally unrelated to this PR), so given the approvals I'm merging.

**Comment by bsipocz** [comment]
Thank you @pllim and @mhvk for the joint effort!

**Pull request merged**